File: WebAPI/PropertyHubAPI/PropertyHubAPI/DTO/Property/PropertyUpdateDto.cs
using PropertyHubAPI.DTO.Addresses;
using PropertyHubAPI.DTO.Features;

namespace PropertyHubAPI.DTO.Property
{
    public class PropertyUpdateDto
    {
        public string? Type { get; set; }
        public decimal? Price { get; set; }
        public int? Bedrooms { get; set; }
        public int? Bathrooms { get; set; }
        public int? Parkings { get; set; }
        public int? Size { get; set; }
        public int? YearBuilt { get; set; }
        public decimal? Tax { get; set; }
        public AddressDto? Address { get; set; }
        public string? Status { get; set; }
        public string? Description { get; set; }
        public string? AgentName { get; set; }
        public List<string>? ImageUrls { get; set; }
        public FeatureDto? Feature { get; set; }
        public required DateTime LastUpdate { get; set; }
    }
}

File: client/src/models/PropertyUpdateDto.ts
import { AddressDto } from './AddressDto';
import { FeatureDto } from './FeatureDto';

export interface PropertyUpdateDto {
    type?: string;
    price?: number;
    bedrooms?: number;
    bathrooms?: number;
    parkings?: number;
    size?: number;
    yearBuilt?: number;
    tax?: number;
    address?: AddressDto;
    status?: string;
    description?: string;
    agentRegistrationNumber?: string;
    imageUrls?: string[];
    feature?: FeatureDto;
    lastUpdate: Date;
}